exonum-java-binding/qa-service/src/test/java/com/exonum/binding/qaservice/Integration.java

```diff
@@ -17,19 +17,20 @@
 package com.exonum.binding.qaservice;
 
 import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
+import org.junit.jupiter.api.Tag;
 
 /**
- * Indicates that it might make sense to promote the annotated element
- * to the core Exonum Java Binding library.
+ * Indicates that a test is an integration test. Tags the annotated test with "integration"
+ * tag.
  */
-@Target({
-    ElementType.TYPE,
-    ElementType.METHOD
-})
-public @interface PromoteToCore {
-  /**
-   * Why the element might be promoted to the core library.
-   */
-  String value() default "";
+@Target({ ElementType.TYPE, ElementType.METHOD })
+@Retention(RetentionPolicy.RUNTIME)
+@Inherited
+@Tag("integration")
+public @interface Integration {
```

Comment:
Why did you go with this annotation instead of `*IntegrationTest.java`?

Reply:
The filtering by file name appeared broken with @Nested tests (<excludes>**/*IntegrationTest.java<excludes> did not exclude the @Nested tests appearing inside *IntegrationTest). Regex by class didn't work either :(

I'll try to put together a reproducing project and submit an issue to surefire project.

Reply:
Turns out, it reproduces only if the `@Nested ` class is names *Tests.

Reply:
https://issues.apache.org/jira/browse/SUREFIRE-1720 

We could use a pattern instead, for higher consistency with other modules.

Reply:
OK, I'll think will do that later: https://jira.bf.local/browse/ECR-3852